### PR TITLE
feat(Global Environments): Introduce a new top level file type called global environments.

### PR DESCRIPTION
**Pull request by gatzjames**
Highlights:

- [x] Adds a new workspace type `environment`
- [x] Collections can now use Global environments alongside collection environments
- [x] Global environments can be synced using Cloud Sync / Git Sync
- [x] Global environments apply on top of Base environments using the same hierarchy model

Next steps:
- [ ] Enable `globals` API in pre-request/after-response scripts
- [ ] Test Import/Export and make sure imports with globals work as expected
- [ ] Environment Settings
- [ ] Review the objects that need to be created for envs (e.g. filter out CookieJar or other stuff that don't make sense)

### New Environment picker UI:
image

### Global Environments in the Project view:
image

### Updates to Models

- Workspace now has a new type `environment`
- A workspace can have an active Global Environment (workspaceMeta has a new property activeGlobalEnvironmentId that references that)

### CODE REVIEW

**Comment by jackkav** [comment]
can you include some screen shots in the description please. I think the ux is a bit contentious. I would also like to see a clear the environment model addressing the current inconsistencies in order to code review this approach.

File: packages/insomnia/src/ui/routes/project.tsx [hunk 1]
```diff
@@ -114,32 +114,36 @@ async function getAllTeamProjects(organizationId: string) {
...
   unsynced: 'cloud-download',
+  environment: 'code',
 };
 
 export const scopeToBgColorMap: Record<string, string> = {
   design: 'bg-[--color-info]',
   collection: 'bg-[--color-surprise]',
   'mock-server': 'bg-[--color-warning]',
   unsynced: 'bg-[--hl-md]',
+  environment: 'bg-[--color-font]',
```
Comment (ihexxa) [inline packages/insomnia/src/ui/routes/project.tsx:138]: nit: Should it be `--color-font` or `color-bg`?

**Review by ihexxa — COMMENTED** [review]
Went through some parts like data model, rendering and the UI part is not checked yet.

File: packages/insomnia/src/ui/routes/project.tsx (hunk 1 above)
Comment (gatzjames) [reply packages/insomnia/src/ui/routes/project.tsx:138]: Yep it's correct we want the color-font for bg here. Agree that it's counter intuitive but there are cases were we need the font color as bg 🫣

**Review by ihexxa — COMMENTED** [review]
Tend to move forward but there are many details in the component so it may deserve more detailed check.

**Review by ihexxa — APPROVED** [review]
LGTM we've discuss several topics like relationship syncing, ux consistency and overall it looks to me. We will also apply it to the scripting feature, have more tests on it and move to the importing/exporting implementation.

**Pull request merged**